<a href="https://colab.research.google.com/github/etorres09311/Python-Assignments/blob/main/Python_literals.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Python literals

## Objective
1. Understand how Python uses literals.
2. Differentiate Integer, Float, Boolean and String data types.
3. Understanding how arithmetic operations are performed and recognizing the precedence of operators.

## Prerequisite

- Python & literals lecture

## What do you need to complete this exercise?

You can perform this exercise in any Python IDE, including JupyterLab or Google Colab.

### Literals

a) What will the output be without using Python? Aim for 80% accuracy.

```print(5 + 2 - 2)```

```print(5 / 2)```

```print(6 // 2) # // integer division```

```print(2. * 3)```

```print(2 < 4)```

```print(2 >= 2)```

```print("Hello"+"World")```

```print("bla" * 3)```

```print(2 * 3 ** 3)```

```print(5 * 25 // 13 + 100 / 2 % 13 // 2)```

```print(2 * 3 % 5)```

```print((2 % -4), (2 % 4), (2 ** 3 ** 2))```


1. 5
2. 2.5
3. 3
4. 6.0
5. True
6. True
7. Hello World
8. bla *3
9. 54
10. 14
11. 1
12. -2, 2, 512

b) Please verify using Python to determine if you have achieved at least 80% correct answers.

In [1]:
print(5 + 2 - 2)

print(5 / 2)

print(6 // 2) # // integer division

print(2. * 3)

print(2 < 4)

print(2 >= 2)

print("Hello"+"World")

print("bla" * 3)

print(2 * 3 ** 3)

print(5 * 25 // 13 + 100 / 2 % 13 // 2)

print(2 * 3 % 5)

print((2 % -4), (2 % 4), (2 ** 3 ** 2))

5
2.5
3
6.0
True
True
HelloWorld
blablabla
54
14.0
1
-2 2 512


### Data type


a) What will the output be without using Python? Aim for 80% accuracy.
```type("Hello")```

```type(1+"2")```

```type(1.)```

```type('A')```

```type(500)```

```type(True)```

```type("False")```


1. Hello
2. Error?
3. 1.0
4. A
5. 500
6. True
7. False

b) Please verify using Python to determine if you have achieved at least 80% correct answers.

In [12]:
type("hello")
type(1.)
type('A')
type(500)
type(True)
type("False")

str

### Operator precedence

a) Write your arithmetic expression that uses at least one operator from each group. For example ```5 * 25 % 13 + 100 / -2 * 13 // 2 ** 3```

In [17]:
5*4**4+20/2-4//8%10

1290.0

b) What will the output be without using Python after writing your arithmetic expression? Use the table below for your reference.

| Priority | Operator                                                     | Notes                    |
| :------- | :----------------------------------------------------------- | :----------------------- |
| 1        | `**`                                                         | Right to left evaluation |
| 2        | `+`, `-` (note: unary operators located next to the right of the power operator bind more strongly) | unary                    |
| 3        | `*`, `/`, `//`, `%`                                          | Left to right evaluation |
| 4        | `+`, `-`                                                     | binary                   |

1290

c) Please verify using Python to determine if you have done it correctly.

In [18]:
5*4**4+20/2-4//8%10

1290.0

## Challenges

Please describe the challenges you faced during the exercise.

My main challenge is figuring out what operand does what, as many of them are so similar looking that I have to constantly recheck my notes to ensure I'm hopefully doing the right thing.